papaya-web/src/main/java/pl/edu/agh/papaya/mappers/AvailabilityMapper.java

```diff
@@ -2,19 +2,32 @@
 
 import java.time.Duration;
 import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 import pl.edu.agh.papaya.api.model.AvailabilityDto;
 import pl.edu.agh.papaya.model.Availability;
+import pl.edu.agh.papaya.service.sprint.SprintService;
 
 @Component
+@RequiredArgsConstructor
 public class AvailabilityMapper implements Mapper<Availability, AvailabilityDto> {
 
+    private final SprintService sprintService;
+
     @Override
     public AvailabilityDto mapToApi(Availability availability) {
+        Long timeAvailable = Optional.ofNullable(availability.getTimeAvailable())
+                .map(Duration::toMinutes)
+                .orElse(0L);
+
+        Double prevAverageSprintCoefficient =
```

Comment:
Hmmm, nie wiem, czy mi się podoba, że mapper ma większą "logikę". A nawet jakby mi się podobało, to nie podoba mi się, że używamy tej metody `mapToApi` na liście batchowo (defaultowa metoda w interfejsie `Mapper`), więc to `prevAverageSprintCoefficient` (zapytanie na bazie) się robi osobno dla każdego `Availability`

Zrobiłabum tutaj (i w `UserAvailabilityMapper` chyba też przy okazji) osobne metody typu
`public AvailabilityDto mapToApi(Availability availability, double coefficient);`
`public List<AvailabilityDto> mapToApi(List<Availability> availability, double coefficient);`
Te metody bez drugiego argumentu by mogły wołać te z drugim argumentem z czynnikiem 1 czy coś (albo w nich mogą się nie ustawiać w ogóle te przeskalowane wartości czy coś - przy zwracaniu zwykłego `AvailabilityDto` użytkownikowi to może mieć więcej sensu). Tę defaultową w interfejsie można zoverride'ować

Reply:
[zebyWKomentarzuNieByloScianyTekstu.txt](https://github.com/papaya-agh/papaya-backend/files/3194235/zebyWKomentarzuNieByloScianyTekstu.txt)

(Napisałam sobie szkic kodu, pisząc ten komentarz, to wklejam, bo nie jestem pewna, czy dobrze wytłumaczyłam, o co mi chodzi)

Reply:
`public AvailabilityDto mapToApi(Availability availability, double coefficient)` nie bo pozwala na łatwe utworzenie rozbieżności pomiędzy rzeczywistym współczynnikiem a tym podanym, mamy dwa źródła prawdy
i nie możemy zakładać domyślnego bo też mamy rozbieżność
to musi być liczone tutaj żeby zachować spójność

Reply:
podziele to, ale zostawie metody jako prywatne